From 7edfbc641d0726a09d97e0383b9fe504b2d1200c Mon Sep 17 00:00:00 2001
From: Robert Marshall <rmarshall@redhat.com>
Date: Mon, 22 Feb 2016 15:30:05 -0500
Subject: [PATCH 80/97] Warn if grub password will not be read (#1290803)

It is possible for a system to have never run grub-mkconfig and add the
section that reads the user.cfg file which contains a user set GRUB
password. Users in that scenario will now be warned that grub-mkconfig
must be run prior to their newly set password taking effect.

Resolves: rhbz#1290803
---
 util/grub-setpassword.in | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/util/grub-setpassword.in b/util/grub-setpassword.in
index 2923f437ebe8..dd7b7932673f 100644
--- a/util/grub-setpassword.in
+++ b/util/grub-setpassword.in
@@ -121,3 +121,8 @@ fi
 install -m 0600 /dev/null "${grubdir}/user.cfg" 2>/dev/null || :
 chmod 0600 "${grubdir}/user.cfg" 2>/dev/null || :
 echo "GRUB2_PASSWORD=${MYPASS}" > "${grubdir}/user.cfg"
+
+if ! grep -q "^### BEGIN /etc/grub.d/01_users ###$" "${grubdir}/grub.cfg"; then
+    echo "WARNING: The current configuration lacks password support!"
+    echo "Update your configuration with @grub_mkconfig@ to support this feature."
+fi
-- 
2.13.0

